Mathlib/FieldTheory/IntermediateField.lean

```diff
@@ -743,6 +743,37 @@ theorem extendScalars_injective :
 
 end ExtendScalars
 
+section Restrict
+
+variable {F E : IntermediateField K L} (h : F ≤ E)
+
+/--
+If `F ≤ E` are two intermediate fields of `L / K`, then `F` is also an intermediate field of
+`E / K`. It is an inverse of `IntermediateField.lift`, and can be viewed as a dual to
+`IntermediateField.extendScalars`.
+-/
+protected def restrict : IntermediateField K E :=
+  (IntermediateField.inclusion h).fieldRange
```

Comment:
I think this need not to be `protected`.

Also, do you think the definition `comap E.val F` is better? This basically is `E ⊓ F` viewed as an intermediate field of `E / K`. But since `F ≤ E`, it is just `F`. Probably this definition will make the proof of `lift_restrict` simpler.

Reply:
Isn't `restrict` too general of a name to not be protected?

Reply:
I don't think so; even `lift` is not protected.